### PR TITLE
release v1.1.0-rc01

### PR DESCRIPTION
**Pull request by vkatz**

### CODE REVIEW

File: tiamat-koin/build.gradle.kts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 plugins {
+    alias(libs.plugins.kotlin.multiplatform)
     alias(libs.plugins.android.library)
     alias(libs.plugins.jetbrains.compose)
-    alias(libs.plugins.kotlin.multiplatform)
     alias(libs.plugins.m2p)
 }
 
-version = "1.0.3"
+version = "1.1.0"
```
Comment (egorikftp) [inline tiamat-koin/build.gradle.kts:8]: `1.1.0-rc01`?

File: tiamat/src/androidMain/kotlin/com/composegears/tiamat/Platform.android.kt [hunk 2]
```diff
@@ -32,4 +32,11 @@ internal actual fun <Args> NavDestinationScope<Args>.PlatformContentWrapper(
...
-}
\ No newline at end of file
+}
+
+/**
+ * We can not call T::class in @Composable functions,
+ *
+ * workaround is to call it outside of @Composable via regular inline fun
+ */
+actual inline fun <reified T : Any> className(): String = T::class.qualifiedName!!
```
Comment (egorikftp) [inline tiamat/src/androidMain/kotlin/com/composegears/tiamat/Platform.android.kt:42]: can be internal?

File: tiamat/src/commonMain/kotlin/com/composegears/tiamat/Utils.kt [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
 /**
- * We can not call T::class in @Composable functions,
- *
- * workaround is to call it outside of @Composable via regular inline fun
+ * @return true if all elements match the given predicate or collection is empty
  */
-inline fun <reified T : Any> className(): String = T::class.qualifiedName!!
\ No newline at end of file
+internal fun <T> Iterable<T>.allIndexed(predicate: (Int, T) -> Boolean): Boolean {
+    if (this is Collection && isEmpty()) return true
```
Comment (egorikftp) [inline tiamat/src/commonMain/kotlin/com/composegears/tiamat/Utils.kt:7]: the whole extension looks the same as `all{}` 

image

what about
```kotlin
internal fun <T> Iterable<T>.allIndexed(predicate: (Int, T) -> Boolean): Boolean {
    return withIndex().all { predicate(it.index, it.value) }
}
```

File: tiamat-koin/build.gradle.kts (hunk 1 above)
Comment (vkatz) [reply tiamat-koin/build.gradle.kts]: fixed

File: tiamat/src/androidMain/kotlin/com/composegears/tiamat/Platform.android.kt (hunk 2 above)
Comment (vkatz) [reply tiamat/src/androidMain/kotlin/com/composegears/tiamat/Platform.android.kt:42]: Public `rememberViewModel` uses `className` as inline fun, so it should be public

File: tiamat/src/commonMain/kotlin/com/composegears/tiamat/Compose.kt [hunk 4]
```diff
@@ -208,7 +201,7 @@ fun Navigation(
     // display current entry + animate enter/exit
     AnimatedContent(
         targetState = navController.currentNavEntry,
-        contentKey = { it?.let { "${it.destination.name}:${it.uuid}" } ?: "" },
+        contentKey = { it?.let { "${it.destination.name}:${it.navId}" } ?: "" },
```
[the remark this answers is not in the document]
Comment (vkatz) [reply tiamat/src/commonMain/kotlin/com/composegears/tiamat/Compose.kt:204]: done

File: tiamat/src/commonMain/kotlin/com/composegears/tiamat/NavController.kt [hunk 5]
```diff
@@ -121,16 +176,11 @@ class NavController internal constructor(
...
-
-    /**
-     * @return current backstack destinations entries list
+     * @return current backstack
      *
-     * @see [NavDestinationEntry]
+     * @see [NavEntry]
      */
-    fun getBackStackEntries() = backStack.map { it.toNavDestinationEntry() }
+    fun getBackStack() = backStack as List<NavEntry<*>>
```
[the remark this answers is not in the document]
Comment (vkatz) [reply tiamat/src/commonMain/kotlin/com/composegears/tiamat/NavController.kt:184]: done

File: tiamat/src/commonMain/kotlin/com/composegears/tiamat/NavEntry.kt [hunk 6]
```diff
@@ -1,100 +1,130 @@
...
+}
+
+/**
+ * Converts [NavDestination] into [NavEntry]
+ *
+ * @param navArgs entry navArgs
+ * @param freeArgs entry freeArgs
+ * @param navResult entry navResult
+ */
+fun <Args> NavDestination<Args>.toNavEntry(
```
[the remark this answers is not in the document]
Comment (vkatz) [reply tiamat/src/commonMain/kotlin/com/composegears/tiamat/NavEntry.kt:121]: intend to be open to be able bild args/no args via `if` or another providable destination logic
```
val r = rememberNavController(
                startDestination = if(cond) MainScreen.toNavEntry() else MainScreen.toNavEntry(args)
            )
```

File: tiamat/src/commonMain/kotlin/com/composegears/tiamat/Utils.kt (hunk 3 above)
Comment (vkatz) [reply tiamat/src/commonMain/kotlin/com/composegears/tiamat/Utils.kt:7]: withIndex() will create a wrapper of iterator and do alot extra work to calc index, 
forEachIndexed under the hood is just `int++` + `simple iteration` - this is more optimized and faster

File: tiamat/src/commonMain/kotlin/com/composegears/tiamat/NavEntry.kt (hunk 6 above)
[the remark this answers is not in the document]
Comment (egorikftp) [reply tiamat/src/commonMain/kotlin/com/composegears/tiamat/NavEntry.kt:121]: maybe we need to add this in documentaion

**Review by egorikftp — APPROVED** [review]

**Pull request merged**